core/src/subject/SubjectEntity.ts

```diff
@@ -8,26 +8,208 @@ export type QueryPartialEntity<T> = {
+      AllLinks
+    ),
+    sort(AllLinks, SortedLinks),
+    SortedLinks = [[${variables.join(", ")}]|_]
+  `;
+}
+
+function buildPropertiesQuery(properties?: string[]): string {
+  // Gets the name, value, and resolve boolean for all (or some) properties on a SubjectEntity instance
+  // If no properties are provided, all are included
+  return `
+    findall([PropertyName, PropertyValue, Resolve], (
+      % Constrain to specified properties if provided
+      ${properties ? `member(PropertyName, [${properties.map((name) => `"${name}"`).join(", ")}]),` : ""}
+
+      % Get the property name and resolve boolean
+      property(SubjectClass, PropertyName),
+      property_getter(SubjectClass, Base, PropertyName, PropertyValue),
+      (property_resolve(SubjectClass, PropertyName) -> Resolve = true ; Resolve = false)
+    ), Properties)
+  `;
+
+  // Works with literal_from_url
+  // return `
+  //   findall([PropertyName, PropertyValue, Resolve], (
+  //     % Constrain to specified properties if provided
+  //     ${properties ? `member(PropertyName, [${properties.map((name) => `"${name}"`).join(", ")}]),` : ""}
+
+  //     % Get the property name, URI, and resolve boolean
+  //     property(SubjectClass, PropertyName),
+  //     property_getter(SubjectClass, Base, PropertyName, PropertyURI),
+  //     (property_resolve(SubjectClass, PropertyName) -> Resolve = true ; Resolve = false),
+
+  //     % Decode the property value from the URI
+  //     literal_from_url(PropertyURI, PropertyValue, _)
+  //   ), Properties)
+  // `;
+}
+
+function buildCollectionsQuery(collections?: string[]): string {
+  // Gets the name and array of values for all (or some) collections on a SubjectEntity instance
+  // If no collections are provided, all are included
+  return `
+    findall([CollectionName, CollectionValues], (
+      % Constrain to specified collections if provided
+      ${collections ? `member(CollectionName, [${collections.map((name) => `"${name}"`).join(", ")}]),` : ""}
+
+      collection(SubjectClass, CollectionName),
+      collection_getter(SubjectClass, Base, CollectionName, CollectionValues)
+    ), Collections)
+  `;
+}
+
+// Todo: greaterThanOrEqualTo, lessThanOrEqualTo
+function buildWhereQuery(where: WhereCondition): string {
+  // Constrains the query to instances that match the provided where conditions
+
+  function formatValue(key, value) {
+    // Base format on type of value by default
+    let isString = typeof value === "string";
+    // Override for special fields
+    if (key === "author") isString = true;
+    if (key === "timestamp") isString = false;
+    // Wrap strings in quotes
+    return isString ? `"${value}"` : value;
+  }
+
+  return (Object.entries(where) as [string, WhereCondition][])
+    .map(([key, value]) => {
+      const isSpecial = ["author", "timestamp"].includes(key);
+      const getter = `property_getter(SubjectClass, Base, "${key}", URI), literal_from_url(URI, V, _)`;
```

Comment:
if the URI is not a literal, I guess literal_from_url won't match the property is omitted

Reply:
should be handled generically with property query code above